src/components/expense-form.tsx

```diff
@@ -449,24 +458,13 @@ export function ExpenseForm({
                                                   participant === id,
                                               )?.shares
                                             }
-                                            onChange={(event) => {
-                                              const value = event.target.value
-                                              const fixedValue =
-                                                typeof value === 'string'
-                                                  ? value.replace(/,/g, '.')
-                                                  : value
-
-                                              return field.onChange(
-                                                field.value.map((p) =>
-                                                  p.participant === id
-                                                    ? {
-                                                        participant: id,
-                                                        shares: fixedValue,
-                                                      }
-                                                    : p,
-                                                ),
+                                            pattern="[0-9]+([,\.][0-9]+)?"
+                                            onChange={(event) =>
```

Comment:
This breaks, because this `field.onChange()` has a different signature than the one above.

May I suggest:

```tsx
onChange={(event) =>
      field.onChange(
        field.value.map((p) =>
          p.participant === id
            ? {
                participant: id,
                shares:
                  enforceCurrencyPattern(
                    event,
                  ),
              }
            : p,
        ),
      )
    }
```

where `enforceCurrencyPattern()` becomes

```tax
const enforceCurrencyPattern = (event: ChangeEvent<HTMLInputElement>) =>
  event.target.value
    // replace first comma with #
    .replace(/[.,]/, '#')
    // remove all other commas
    .replace(/[.,]/g, '')
    // change back # to dot
    .replace(/#/, '.')
    // remove all non-numeric and non-dot characters
    .replace(/[^\d.]/g, '')
```

Note: this also solves the issue where text could be entered with multiple comas